toptal/picasso
# test: page in various screens

[FX-4090]

### Description

This pull request adds tests that make sure that the dropdown menu (with hamburger button) works and looks as expected on all screens.

**⚠️ Additional testing was performed**

The `disablePortal` was added to the `packages/picasso/src/PageHamburger/PageHamburger.tsx` in order to observe the menu when testing in Cypress (fix discovered by @mkrl)

### How to test

- Compare https://picasso.toptal.net/fx-4090-add-all-breakpoint-ranges-as-targets-in-pagetopbar-test/?path=/story/layout-page--page and https://picasso.toptal.net/?path=/story/layout-page--page on various screen sizes
- (Client Portal testing) Log in as Primary Contact for any Client (for example, for https://staff-portal.toptal.net/clients/295868) on staging, compare https://portal.toptal.net/ and https://client-portal-pr-6960.toptal.rocks/ on various screens
- (Staff Portal testing) Please see the https://staff-portal-pr-10698.toptal.rocks (https://github.com/toptal/staff-portal/pull/10698), but the change should not affect portal as it it uses custom menu
- (Talent Portal testing) The Portal has not migrated to new breakpoints, so the following can be done to test it locally
  - log in as active Talent (for example, as https://staff-portal.toptal.net/talents/3747845) at staging
  - check out `master` branch, set up default configuration
  - go to `node_modules/@toptal/picasso/PageHamburger/PageHamburger.js` file and add `disablePortal: true`
  
<img width="964" alt="Screenshot 2023-07-24 at 21 09 38" src="https://github.com/toptal/picasso/assets/1390758/2f3c3b4c-24fb-49ae-b84d-6b1c3f9d1b15">

  - run portal locally and check it on various screen sizes
 
### Development checks

- [N/A] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [N/A] Annotate all `props` in component with documentation
- [N/A] Create `examples` for component
- [N/A] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Remove testing pull requests and branches before merge (https://github.com/toptal/staff-portal/pull/10698 and https://github.com/toptal/client-portal/pull/6960)

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>


[FX-4090]: https://toptal-core.atlassian.net/browse/FX-4090?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
